### PR TITLE
Fix: support models saved with mlflow=2 in mlflowserver conda_env_create

### PR DESCRIPTION
**Pull request by maxb2**
**What this PR does / why we need it**:

This PR supports parsing `MLModel` files that were saved using `mlflow=2.* `. They introduced subfields for the environment in https://github.com/mlflow/mlflow/pull/6684. 

Here is an example of the differences between the `MLModel` files:

```yaml
### MLModel saved with mlflow=1
artifact_path: model
flavors:
  python_function:
    env: conda.yaml
    loader_module: mlflow.sklearn
    model_path: model.pkl
    predict_fn: predict
    python_version: 3.8.15
  sklearn:
    code: null
    pickled_model: model.pkl
    serialization_format: cloudpickle
    sklearn_version: 1.1.3
mlflow_version: 1.30.0
model_uuid: 9af999cad24d4f7e855ef3073a3f5163
run_id: 340709c027e3424bbe0e13e4803cad46
utc_time_created: '2022-12-02 15:47:51.018732'
```

```yaml
### MLModel saved with mlflow=2
artifact_path: model
flavors:
  python_function:
    env:
      conda: conda.yaml
      virtualenv: python_env.yaml
    loader_module: mlflow.sklearn
    model_path: model.pkl
    predict_fn: predict
    python_version: 3.8.15
  sklearn:
    code: null
    pickled_model: model.pkl
    serialization_format: cloudpickle
    sklearn_version: 1.1.3
mlflow_version: 2.0.1
model_uuid: 9649ad4b286e4b4fb52843a8c7146f64
run_id: fedf638f71a944eea4bf717008961a99
utc_time_created: '2022-12-09 20:26:51.455742'
```

**Which issue(s) this PR fixes**:

I didn't see any related issues and the fix was easy enough to go directly to a PR.

**Special notes for your reviewer**:
I wanted to read `mlflow_version` from the `MLModel` file, but that field was only introduced in `mlflow=1.25` and the image is pinned to `mlflow<1.24.0`. So, I just added a check to see if the `env` field is a dictionary and if so, assumed that I would need to get the `conda` field.

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @maxb2. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **sachinvarghese**
You can assign the PR to them by writing `/assign @sachinvarghese` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jian-mo — APPROVED** [review]

**Review by adriangonz — APPROVED** [review]
Hey @maxb2 ,

Thanks a lot for taking the lead on this one, and apologies for the delay. 

Changes look good and integration tests seem to pass, so this should be good to go :+1:

**Comment by cliveseldon** [comment]
We are working on a 1.16.0 release as you can see [here](https://github.com/SeldonIO/seldon-core/projects/47) which will contain this fix amongst other updates.

**Comment by edfincham** [comment]
That's brilliant to hear - do you have a set cadence for these releases or is it "ready when it's ready"?

**Comment by edfincham** [comment]
Thanks - I'll expect it when I see it :smile: 

In the meantime, do you publish pre-release images?

**Comment by cliveseldon** [comment]
Yes see `latest` on DockerHub

**Comment by cliveseldon** [comment]
We don't do nightly build pushes at present.

**Pull request merged**
